rclcpp/test/rclcpp/test_memory_strategy.cpp

```
@@ -228,6 +232,11 @@ TEST_F(TestMemoryStrategy, get_client_by_handle) {
     EXPECT_EQ(
       nullptr,
       memory_strategy()->get_client_by_handle(client_handle, weak_groups_to_nodes));
+    // Clear all handles to cleanup when node is destroyed
+    memory_strategy()->clear_handles();
+    // reset client handle before node destruction, otherwise it tries to access node
+    // via weak_ptr in its deleter
+    client_handle.reset();
```

nit: I was wondering if `memory_strategy()->clear_handles()` could be put in the `TearDown` method to cut down on the test repetition. Would the order of handle clearing and client handle destruction matter for this test?

---

@skyegalaxy thanks for reviewing this.

the thing is that `memory_strategy()->clear_handles()` needs to be issued before node goes out of the scope. that actually suppress the error message https://github.com/ros2/rclcpp/issues/2968.

---

Fair enough. LGTM!